slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbSlashingProtection.java

```diff
@@ -100,12 +107,31 @@ public void export(final OutputStream output) {
     }
   }
 
+  @Override
+  public void registerGenesisValidatorsRoot(final Bytes32 genesisValidatorsRoot) {
+    jdbi.useTransaction(
+        SERIALIZABLE,
+        h -> {
+          final Optional<Bytes32> dbGvr = metadataDao.findGenesisValidatorsRoot(h);
+          final Optional<Boolean> isValidGvr = dbGvr.map(gvr -> gvr.equals(genesisValidatorsRoot));
```

Comment:
nit: putting the orElse on this line would make it easier to read:
final boolean gvrMatchesDb = dbGvr.map(gvr -> gvr.equals(genesisValidatorRoot).orElse(false);
if(!gvrMatchesDb) .....

Reply:
kinda surprised this same logic can't be used in the maySignBlock and maySignAttestation

Reply:
actually - seems duplicated, with different error msg